src/NetworkManager.cpp

```diff
@@ -97,24 +106,74 @@ bool NetworkManager::IsUrlAllowed(const std::string& url)
 
 HttpRequestFuturePtr NetworkManager::HttpRequest(const HttpRequestArgs& args)
 {
-	ix::HttpRequestArgsPtr req = this->httpClient.createRequest(args.url, args.method);
+	auto &client = args.downloadFile.empty() ? this->httpClient : this->downloadClient;
+	auto downloadFile = std::make_shared<RageFile>();
+	std::string downloadFilename;
+
+	ix::HttpRequestArgsPtr req = client.createRequest(args.url, args.method);
 	req->body = args.body;
 	req->multipartBoundary = args.multipartBoundary;
 
+	req->extraHeaders["User-Agent"] = this->getUserAgent();
 	for (const auto& entry : args.headers)
 	{
 		req->extraHeaders[entry.first] = entry.second;
 	}
 
-	if (args.connectTimeout > 0) {
-		req->connectTimeout = args.connectTimeout;
+	if (!args.downloadFile.empty())
+	{
+		downloadFilename = std::string("/Downloads/") + args.downloadFile;
+		if (!downloadFile->Open(downloadFilename, RageFile::WRITE | RageFile::STREAMED))
+		{
+			req->cancel = true;
```

Comment:
I opted to not write a log message, because we pass the error to the response callback. It's up to the user to decide whether that should be logged or not.
```C++
		if (downloadFile->IsOpen())
		{
			RString error = downloadFile->GetError();
			downloadFile->Close();

			if (!error.empty())
			{
				std::string errorMessage = "could not write to " + downloadFile->GetPath() + ": " + error;
				if (args.onFileError)
					args.onFileError(errorMessage);

				FILEMAN->Remove(downloadFilename);
				return;
			}
		}
```

Reply:
Ahh I see. Okay!